rfcs/0000_monty_versioning.md

```diff
@@ -0,0 +1,120 @@
+- Start Date: 2025-02-20
+- RFC PR: (leave this empty, it will be filled in after RFC is merged)
+
+# Summary
+
+Begin versioning Monty using [semantic versioning](https://semver.org/).
+
+# Motivation
+
+We want papers published by the Thousand Brains Project to be reproducible. The likely approach the project will adopt is to create a repository for each paper and include the code to reproduce the paper in the repository. When the paper relies on `tbp.monty`, it should be clear which version of `tbp.monty` is needed to reproduce the paper. Ideally, the paper repository declares a specific `tbp.monty` version as a dependency that is installed like any other dependency.
+
+# Explanation
+
+Monty uses [semantic versioning](https://semver.org/) for versions.
+
+The intent behind semantic versioning is to provide a way to rapidly and tersely communicate to people and machines the compatibility of a software change.
+
+In summary, the version number is `MAJOR.MINOR.PATCH`.
+
+* `MAJOR` is incremented when backwards-incompatible (breaking) changes are made.
+* `MINOR` is incremented when new functionality is added in a backwards-compatible manner.
+* `PATCH` is incremented when backwards-compatible bug fixes are made.
+* CI, tools, infrastructure, documentation, or RFC changes do not increment the version number.
+
+The current version can be found in [`src/tbp/monty/__init__.py`](https://github.com/thousandbrainsproject/tbp.monty/blob/main/src/tbp/monty/__init__.py) as the `__version__` variable.
+
+## Updating the version
+
+### Constraints
+
+Here are some constraints related to updating the version:
+
+* We want the current version to be in the source code.
+* The `main` branch is protected and changes can be pushed only via Pull Requests.
+
```

Comment:
I would add a constraint that it needs to be programmatically accessible for things like the print_version tool. https://github.com/thousandbrainsproject/tbp.monty/tree/main/tools/print_version

Reply:
Added a constraint "We want the current version to be programmatically accessible, e.g., by the `tools.print_version.cli` tool." in https://github.com/thousandbrainsproject/tbp.monty/pull/186/commits/fb95a0c214f8f2ee028a21192cb40d4703ef0e55